regexp/regexp.go

```diff
@@ -0,0 +1,243 @@
+	return starlark.Bool(re.MatchString(src)), nil
+}
+
+func find(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return starlark.String(re.FindString(src)), nil
+}
+
+func findAll(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindAllString(src, max)), nil
+}
+
+func findSubmatches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var src string
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.FindStringSubmatch(src)), nil
+}
+
+func replaceAll(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src  string
+		repl starlark.Value
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &src, &repl); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	switch x := repl.(type) {
+	case starlark.Callable:
+		var fnErr error
+		fn := func(matched string) string {
+			res, err := starlark.Call(thread, repl, starlark.Tuple{starlark.String(matched)}, nil)
+			if err != nil {
+				// Save the error to be able to return it to the caller
+				fnErr = err
+				return ""
+			}
+			resp, ok := res.(starlark.String)
+			if !ok {
+				// Save the error to be able to return it to the caller
+				fnErr = fmt.Errorf("%s returned %s, want string", x.Name(), res.Type())
+				return ""
+			}
+			return string(resp)
+		}
+		result := re.ReplaceAllStringFunc(src, fn)
+		if fnErr != nil {
+			return nil, fnErr
+		}
+		return starlark.String(result), nil
+	case starlark.String:
+		return starlark.String(re.ReplaceAllString(src, convertReplacementPattern(string(x)))), nil
+	}
+	return nil, fmt.Errorf("got %s, want a string or callable", repl.Type())
+}
+
+func split(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return nil, err
+	}
+
+	re := b.Receiver().(*Regexp).re
+	return toList(re.Split(src, max)), nil
+}
+
+// Converts the replacement pattern to make it compatible with what the regexp package expects.
+func convertReplacementPattern(repl string) string {
+	// Escapes the specifal dollar characters if any
+	repl = strings.ReplaceAll(repl, "$", "$$")
```

Comment:
Does this always work? I think it's over-zealous, and would break this example:

```
import re
s = 'big$money'
re.match(r'[\w]+\$[\w]+', s)
```

Rewriting a regex to work with a different syntax probably requires fully parsing it, not just single character replacement. Of course, it's a bit of an edge case, and if we're ok with losing correctness until the upstream version is done, it's probably not too big of a problem.